source/Ship.cpp

```diff
@@ -129,6 +129,24 @@ namespace {
 		else
 			stat = max(0., .99 * stat);
 	}
+
+	// Get an overview of how many weapon-outfits are equipped.
+	map<const Outfit *, int> GetEquipped(const vector<Hardpoint> &weapons)
+	{
+		map<const Outfit *, int> equipped;
+		for(const Hardpoint &hardpoint : weapons)
+			if(hardpoint.GetOutfit())
+				++equipped[hardpoint.GetOutfit()];
+		return equipped;
+	}
+
+	void LogWarning(const string &modelName, const string &name, const string &warning)
```

Comment:
Since these strings are to be copied anyway:
```suggestion
	void LogWarning(string modelName, string name, string &&warning)
```
And then use std::move in the function body.